src/mmw/apps/bigcz/clients/usgswqp/search.py

```diff
@@ -0,0 +1,129 @@
+    return Point(lng, lat)
+
+
+def parse_record(record):
+    geom = parse_geom(record)
+
+    links = []
+    monitoring_description = record['MonitoringLocationDescriptionText']
+
+    return USGSResource(
+        id=record['MonitoringLocationIdentifier'],
+        title=record['MonitoringLocationName'],
+        description=monitoring_description,
+        author=None,
+        links=links,
+        created_at=None,
+        updated_at=None,
+        geom=geom,
+        details_url='https://www.waterqualitydata.us/provider/{prov}/{org}/{id}/'.format(prov=record['ProviderName'],  # NOQA
+                                                                                         org=record['OrganizationIdentifier'],  # NOQA
+                                                                                         id=record['MonitoringLocationIdentifier']),  # NOQA
+        sample_mediums=None,
+        variables=None,
+        service_org=record['OrganizationIdentifier'],
+        service_orgname=record['OrganizationFormalName'],
+        service_code=record['MonitoringLocationIdentifier'],
+        service_url='https://www.waterqualitydata.us/data/Result/search?siteid={}&mimeType=csv&sorted=no&zip=yes'.format(record['MonitoringLocationIdentifier']),  # NOQA
+        service_title=None,
+        service_citation='We thank the water quality portal.',
+        begin_date=None,
+        end_date=None,
+        monitoring_type=record['MonitoringLocationTypeName'],
+        provider_name=record['ProviderName']
+    )
+
+
+def search(**kwargs):
+    bbox = kwargs.get('bbox')
+    # Currently not being used
+    # to_date = kwargs.get('to_date')
+    # from_date = kwargs.get('from_date')
+
+    if not bbox:
+        raise ValidationError({
+            'error': 'Required argument: bbox'})
+
+    bbox_area = bbox.area() * SQKM_PER_SQM
+
+    if bbox_area > USGS_MAX_SIZE_SQKM:
+        raise ValidationError({
+            'error': 'The selected area of interest with a bounding box of {} '
+                     'km² is larger than the currently supported maximum size '
+                     'of {} km².'.format(round(bbox_area, 2), USGS_MAX_SIZE_SQKM)})  # NOQA
+    params = {
+        # bBox might be used in the future
+        # 'bBox': '{0:.3f},{1:.3f},{2:.3f},{3:.3f}'.format(bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax),  # NOQA
+        'huc': unique_huc12s_in(kwargs.get('geojson')),
+        'mimeType': 'csv',
+        'sorted': 'no',
+        'minresults': '1',
+        'zip': 'yes'
+    }
+
+    try:
+        response = requests.get(CATALOG_URL, params=params)
+        with ZipFile(BytesIO(response.content)) as z:
+            data = csv.DictReader(z.open(z.filelist[0].filename))
+    except requests.Timeout:
+        raise RequestTimedOutError()
+
+    if not data:
+        raise ValueError('Could not fetch data from USGS WQP portal.')
+
+    results = [parse_record(row) for row in data]
+
+    return ResourceList(
+        api_url=None,
```

Comment:
By adding `response.url` here we can ensure that users of the API know the exact third-party endpoint to use themselves if they ever want direct access:

```diff
diff --git a/src/mmw/apps/bigcz/clients/usgswqp/search.py b/src/mmw/apps/bigcz/clients/usgswqp/search.py
index b9de48d1..6fef7a67 100644
--- a/src/mmw/apps/bigcz/clients/usgswqp/search.py
+++ b/src/mmw/apps/bigcz/clients/usgswqp/search.py
@@ -123,7 +123,7 @@ def search(**kwargs):
     results = [parse_record(row) for row in data]
 
     return ResourceList(
-        api_url=None,
+        api_url=response.url,
         catalog=CATALOG_NAME,
         count=len(results),
         results=results)
```

![image](https://user-images.githubusercontent.com/1430060/39944979-b018e220-5536-11e8-8334-e92a40b2d31b.png)